pkg/config/config.go

```diff
@@ -195,13 +192,28 @@ func evalFuncFile() function.Function {
 	})
 }
 
-// LoadComponentConfigBody returns nil if no component with the given
-// name is found in the configuration
-func (c *Config) LoadComponentConfigBody(componentName string) *hcl.Body {
-	for _, component := range c.RootConfig.Components {
-		if componentName == component.Name {
-			return &component.Config
+// lokocfgFiles returns a slice containing the paths of all .lokocfg files found under path p. If p
+// is not a directory, p is returned by itself in the slice.
+func lokocfgFiles(p string) ([]string, error) {
+	isDir, err := util.PathIsDir(p)
+	if err != nil {
+		return nil, fmt.Errorf("failed to stat config path %q: %w", p, err)
+	}
+
+	var paths []string
+
+	if isDir {
+		globPattern := filepath.Join(p, "*.lokocfg")
+
+		configFiles, err := filepath.Glob(globPattern)
+		if err != nil {
+			return nil, fmt.Errorf("bad filepath glob pattern %q: %w", globPattern, err)
 		}
+
+		paths = append(paths, configFiles...)
+	} else {
+		paths = append(paths, p)
 	}
-	return nil
+
+	return paths, nil
 }
```

Comment:
This can be simplified with
```go
if !isDir {
  return []string{p}
}

glob etc...
```

Reply:
Yup, and we also don't need `append()`. Fixed.